examples/pytorch/dimenet/main.py

```diff
@@ -0,0 +1,254 @@
+                        num_blocks=model_params['num_blocks'],
+                        num_bilinear=model_params['num_bilinear'],
+                        num_spherical=model_params['num_spherical'],
+                        num_radial=model_params['num_radial'],
+                        cutoff=model_params['cutoff'],
+                        envelope_exponent=model_params['envelope_exponent'],
+                        num_before_skip=model_params['num_before_skip'],
+                        num_after_skip=model_params['num_after_skip'],
+                        num_dense_output=model_params['num_dense_output'],
+                        num_targets=len(model_params['targets']),
+                        output_init=model_params['output_init']).to(device)
+    elif model_name == 'dimenet++':
+        model = DimeNetPP(emb_size=model_params['emb_size'],
+                          out_emb_size=model_params['out_emb_size'],
+                          int_emb_size=model_params['int_emb_size'],
+                          basis_emb_size=model_params['basis_emb_size'],
+                          num_blocks=model_params['num_blocks'],
+                          num_spherical=model_params['num_spherical'],
+                          num_radial=model_params['num_radial'],
+                          cutoff=model_params['cutoff'],
+                          envelope_exponent=model_params['envelope_exponent'],
+                          num_before_skip=model_params['num_before_skip'],
+                          num_after_skip=model_params['num_after_skip'],
+                          num_dense_output=model_params['num_dense_output'],
+                          num_targets=len(model_params['targets']),
+                          extensive=model_params['extensive'],
+                          output_init=model_params['output_init']).to(device)
+    else:
+        raise ValueError(f'Invalid Model Name {model_name}')
+
+    if pretrain_params['flag']:
+        torch_path = pretrain_params['path']
+        target = model_params['targets'][0]
+        model.load_state_dict(torch.load(f'{torch_path}/{target}.pt'))
+
+        logger.info('Testing with Pretrained model')
+        predictions, labels = evaluate(device, model, test_loader)
+        test_mae = mean_absolute_error(labels, predictions)
+        logger.info(f'Test MAE {test_mae:.4f}')
+
+        return
+    # define loss function and optimization
+    loss_fn = nn.L1Loss()
+    opt = optim.Adam(model.parameters(), lr=train_params['lr'], weight_decay=train_params['weight_decay'], amsgrad=True)
+    scheduler = optim.lr_scheduler.StepLR(opt, train_params['step_size'], gamma=train_params['gamma'])
+
+    # model training
+    best_mae = 1e9
+    no_improvement = 0
+    
+    # EMA for valid and test
+    logger.info('EMA Init')
+    ema_model = copy.deepcopy(model)
+    for p in ema_model.parameters():
+        p.requires_grad_(False)
+    
+    best_model = copy.deepcopy(ema_model)
```

Comment:
I remember you said that the author did not use early stopping. `best_model` is not used anywhere.